dcos/util.py

```diff
@@ -77,6 +77,21 @@ def temptext():
         shutil.rmtree(path, ignore_errors=True)
 
 
+@contextlib.contextmanager
+def path(path):
```

Comment:
Let's name this something else. Maybe `remove_path_on_error` so you can do `with remove_path_on_error(path) as path:`